src/components/Snapshots.vue

```diff
@@ -369,6 +369,16 @@ export default class Snapshots extends Vue {
   getBasicDownloadParams() {
     let w = store.dataset!.width;
     let h = store.dataset!.height;
+
+    // Determine fileName
+    const datasetName = store.dataset?.name || "unknown";
+    const configurationName = store.configuration?.name || "unknown";
+    const snapshotName = this.newName || "unknown";
```

Comment:
When there isn't a snapshot name, we might just want to leave this component out of the filename, rather than have the `unknown` present.  @arjunrajlab Do you have a preference?

Reply:
Agreed, probably better not to have `unknown`.